include/swift/Basic/ByteTreeSerialization.h

```diff
@@ -206,10 +217,10 @@ class ByteTreeWriter {
   write(const T &Object, unsigned Index) {
     validateAndIncreaseFieldIndex(Index);
 
-    auto ObjectWriter = ByteTreeWriter(StreamWriter);
-    ObjectWriter.setNumFields(ObjectTraits<T>::numFields(Object));
+    auto ObjectWriter = ByteTreeWriter(StreamWriter, UserInfo);
+    ObjectWriter.setNumFields(ObjectTraits<T>::numFields(Object, UserInfo));
 
-    ObjectTraits<T>::write(ObjectWriter, Object);
+    ObjectTraits<T>::write(ObjectWriter, Object, UserInfo);
```

Comment:
I prefer adding `ByteTreeWriter::getUserInfo()` like `JSONSerialization`. I believe it's due to `numFields()`, but still... What if `ScalarTraits` types want to use `UserInfo`?

Reply:
My idea was that we pass `UserInfo` into those methods where needed and would like to keep it consistent between `numFields` and `write`. I think that's more important that keeping it in sync with `JSONSerialization` for now.

To this point I had no need for `UserInfo` in `ScalarTraits` and have thus not added it as a parameter. Partly because I didn't want to risk adding an unnecessary performance increase (however small) by passing an unnecessary parameter. Once needed, I think it can be quickly added but I can also add it now if you would prefer.